contracts/src/RIP7755Verifier.sol

```diff
@@ -0,0 +1,83 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.23;
+
+import {IPrecheckContract} from "./IPrecheckContract.sol";
+import {CrossChainCall, FulfillmentInfo} from "./RIP7755Structs.sol";
+
+/// @title RIP7755Verifier
+///
+/// @author Coinbase (https://github.com/base-org/7755-poc)
+///
+/// @notice A Verification contract within RIP-7755.
+///
+/// This contract's sole purpose is to route requested transactions on destination chains and store record of their fulfillment.
+contract RIP7755Verifier {
+    error RIP7755Verifier__InvalidChainId();
```

Comment:
```suggestion
    error InvalidChainId();
```
these will be name spaced under the contract, automatically?

Reply:
For unit testing, sure but could there be some clarity issues once live if this is thrown during a multi-contract txn that fails? i.e. if Contract A also has an `InvalidChainId` error and Contract A calls into the verifier contract, might be confusing on where exactly the error came from. I guess with tools like tenderly (assuming both contracts are verified) then debugging that scenario is fine

Reply:
hmm yeah I'm trying to think of a case where you would decode the error but not know which contract threw it 🤔